File: c/vowel_count/vowel_count.c
#include <stddef.h>
#include <stdio.h>

size_t get_count(const char *s) {

    size_t count = 0;
    const char *vowels = "aeiou";

    while (*s != '\0') {
        const char *v = vowels;
        while (*v != '\0') {
            if (*v == *s) {
                count++;
                break;
            }
            v++;
        }
        s++;
    }

    return count;
}

int main(void) {

    printf("%lu\n", get_count("aoeuhtn"));

    return 0;
}

File: rust/vowel_count/vowel_count.rs
fn main() {
    println!("{}", get_count("Hello world!"));
}

fn get_count(string: &str) -> usize {
    let mut vowels_count: usize = 0;
    let vowels = "aeiou";

    for c in string.chars() {
        for v in vowels.chars() {
            if c == v {
                vowels_count += 1;
            }
        }
    }

    vowels_count
}
